entities/node.cpp

```diff
@@ -42,4 +43,23 @@ std::shared_ptr<Node> Node::getChild(const std::string& childName) const {
     return nullptr;
 }
 
+std::vector<std::shared_ptr<Component>> Node::components() const {
```

Comment:
If you want to output components in sorted order, then it's better to sort node components inside the node logger function.

In the implementation code usually we wouldn't need sorted components and since sorting is taking some computational resources, then let's sorted components only inside logger